### PR TITLE
Frontend/banner

### PR DESCRIPTION
**Pull request by YuanshuoDu**
### Description

Add banner

Fixes # (issue)

### Type of change

Please select the option that best describes the changes made:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Changes

Added banner and optimise the position

### CODE REVIEW

**Comment by 1Solon** [comment]
@Stevendu666 why use an [alert](https://ui.shadcn.com/docs/components/alert) rather then an [alert dialogue?](https://ui.shadcn.com/docs/components/alert-dialog)

**Review by 1Solon — CHANGES_REQUESTED** [review]
Please fix the linting issues

File: Frontend/src/app/terms/page.tsx
Comment (1Solon) [inline Frontend/src/app/terms/page.tsx:1]: This should be split this off into a new PR?

File: Frontend/src/app/signup/page.tsx
Comment (1Solon) [inline Frontend/src/app/signup/page.tsx:1]: This should be split off into the same PR as terms

Comment (YuanshuoDu) [reply Frontend/src/app/signup/page.tsx:1]: has merged the terms into the banner

**Comment by YuanshuoDu** [comment]
> Hey @YuanshuoDu we have moved away from the cookie based auth while your code here checks for `magpie_auth` cookie. Can we change that please?
> 
> ```ts
> useEffect(() => {
>     if (isPageLoaded) {
>       // get cookie
>       const cookieString = document.cookie;
>       const cookies = cookieString.split('; ').reduce((acc, cookie) => {
>         const [name, value] = cookie.split('=');
>         acc[name] = value;
>         return acc;
>       }, {} as Record<string, string>);
> 
>       // check if cookie exists or expired
>       const hasCookie = cookies['magpie_auth'];
>       const expiryTime = cookies['magpie_auth_expiry']; 
> 
>       if (hasCookie && expiryTime) {
>         const expiryDate = new Date(parseInt(expiryTime, 10));
>         const now = new Date();
>         if (now < expiryDate) {
>           setShow(false);
>         } else {
>           setShow(true);
>         }
>       } else {
>         setShow(true);
>       }
>     }
>   }, [isPageLoaded])
> ```

has called new functions instead of old cookies

**Comment by YuanshuoDu** [comment]
> @Stevendu666 why use an [alert](https://ui.shadcn.com/docs/components/alert) rather then an [alert dialogue?](https://ui.shadcn.com/docs/components/alert-dialog)

image
the alert dialog need the trigger and it also will set the background into blur

File: Frontend/src/app/signup/page.tsx
Comment (YuanshuoDu) [reply Frontend/src/app/signup/page.tsx:1]: Have splitted

File: Frontend/src/app/terms/page.tsx
Comment (YuanshuoDu) [reply Frontend/src/app/terms/page.tsx:1]: Have splitted into terms pr

**Comment by YuanshuoDu** [comment]
> @Stevendu666 why use an [alert](https://ui.shadcn.com/docs/components/alert) rather then an [alert dialogue?](https://ui.shadcn.com/docs/components/alert-dialog)

image
has implemented new UI instead of using alert

File: Frontend/src/app/components/banner/CookieConsent.tsx [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+export default function CookieConsent({
+  variant = "default",
+  onAcceptCallback = async () => {
+    await setCookiesAccepted();
+    await commitSessionToCookies();
+  },
+  onDeclineCallback = async () => {
+    try {
+      await deleteSessionFromCookies();
+      console.log("User denied cookie consent");
```
Comment (1Solon) [inline Frontend/src/app/components/banner/CookieConsent.tsx:24]: It's not necessary to log this to console

File: Frontend/src/app/components/banner/CookieConsent.tsx [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    } catch (error) {
+      console.error("Error during handling deny:", error);
```
Comment (1Solon) [inline Frontend/src/app/components/banner/CookieConsent.tsx:26]: Here also

File: Frontend/src/app/components/banner/CookieConsent.tsx [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    }
+  },
+}) {
+  const [isOpen, setIsOpen] = useState(false);
+  const [hide, setHide] = useState(false);
+
+  const accept = async () => {
+    setIsOpen(false);
+    document.cookie =
+      "cookieConsent=true; expires=Fri, 31 Dec 9999 23:59:59 GMT";
```
Comment (1Solon) [inline Frontend/src/app/components/banner/CookieConsent.tsx:36]: We have an existing cookie to track whether the cookie was accepted- this is triggered by the `setCookiesAccepted();` you called earlier.  

Additionally, hard-coding an expiry value like this potentially opens us to issues. Please coordinate with the team if you plan on editing cookie logic in future like this.

File: Frontend/src/app/components/banner/CookieConsent.tsx [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    setTimeout(() => {
+      setHide(true);
+    }, 700);
+    await onDeclineCallback();
+  };
+
+  useEffect(() => {
+    try {
+      setIsOpen(true);
+      if (document.cookie.includes("cookieConsent=true")) {
```
Comment (1Solon) [inline Frontend/src/app/components/banner/CookieConsent.tsx:54]: This should likely be `getCookiesAccepted`, we have predefined behaviour to check if the cookie has already been done. Wherever possible, we should aim not to duplicate code.

**Review by 1Solon — CHANGES_REQUESTED** [review]
image

The cookie banner should probably be further right then it is currently, as it conflicts with the onboarding button

**Comment by YuanshuoDu** [comment]
> image
> 
> The cookie banner should probably be further right then it is currently, as it conflicts with the onboarding button

using the percentage value to left instead of set value to avoid different screen showing different

File: Frontend/src/app/components/banner/CookieConsent.tsx (hunk 1 above)
Comment (YuanshuoDu) [reply Frontend/src/app/components/banner/CookieConsent.tsx]: Deleted

File: Frontend/src/app/components/banner/CookieConsent.tsx (hunk 2 above)
Comment (YuanshuoDu) [reply Frontend/src/app/components/banner/CookieConsent.tsx]: Deleted as well

File: Frontend/src/app/components/banner/CookieConsent.tsx (hunk 3 above)
Comment (YuanshuoDu) [reply Frontend/src/app/components/banner/CookieConsent.tsx]: Deleted hard-coding, called setCookiesAccepted, commitSessionToCookies

File: Frontend/src/app/components/banner/CookieConsent.tsx (hunk 4 above)
Comment (YuanshuoDu) [reply Frontend/src/app/components/banner/CookieConsent.tsx]: replaced to getCookiesAccepted

**Review by 1Solon — APPROVED** [review]

**Pull request merged**